vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc

```diff
@@ -65,31 +64,242 @@ class PromotionTest : public testing::Test {
+
+  promotion_->SetLastCheckTimerIdForTesting(1);
+
+  // Act
+  promotion_->Refresh(false);
+}
+
+TEST_F(PromotionTest, TestRefreshWithRetryWithTimerId) {
+  // Arrange
+  EXPECT_CALL(*mock_ledger_impl_, SetTimer(_, _)).Times(0);
+
+  promotion_->SetLastCheckTimerIdForTesting(1);
+
+  // Act
+  promotion_->Refresh(true);
+}
+
+TEST_F(PromotionTest, TestRefreshWithoutRetryOrTimerId) {
+  // Arrange
+  EXPECT_CALL(*mock_ledger_impl_, SetTimer(_, _)).Times(1);
+
+  promotion_->SetLastCheckTimerIdForTesting(0);
+
+  // Act
+  promotion_->Refresh(false);
+}
+
+TEST_F(PromotionTest, TestRefreshWithRetryWithoutTimerId) {
+  // Arrange
+  EXPECT_CALL(*mock_ledger_impl_, SetTimer(_, _)).Times(1);
+// calls ledger_->SetTimer with start_timer_in as a random value (how to test?)
+
+  promotion_->SetLastCheckTimerIdForTesting(0);
+
+  // Act
+  promotion_->Refresh(true);
+}
+
+// void OnTimer(const uint32_t timer_id);
+// Nb. Cannot test without mocking Promotion
+//   discussed with tmancey and no value as code may be changing
+//   need to confirm with nejc
+// SUCCESS: timer_id == last_check_timer_id_;
+// - expect call Fetch
+// - last_check_timer_id_ = 0
+// SUCCESS: timer_id == retry_timer_id_;
+// - expect call ledger_->GetAllPromotions
+// OR expect call Promotion::Retry
+// FAIL: timer_id != (last_check_timer_id_ or retry_timer_id_)
+// - neither expected (above) to be called
+
+
+// void ClaimTokens(
+//     ledger::PromotionPtr promotion,
+//     ledger::ResultCallback callback);
+TEST_F(PromotionTest, TestClaimTokensWithNullPromotion) {
+  // Arrange
+  bool callback_called = false;
+  ledger::ResultCallback callback =
+      std::bind(
+        [&callback_called](ledger::Result result) {
+          callback_called = (result == ledger::Result::LEDGER_ERROR);
+        }, _1);
+
+  // Act
+  promotion_->ClaimTokens(nullptr, callback);
+
+  // Assert
+  EXPECT_TRUE(callback_called);
+}
+
+TEST_F(PromotionTest, TestClaimTokensWithPromotion) {
+  // Arrange
+//  EXPECT_CALL(*mock_ledger_impl_, LoadURL(_, _, _, _, _, _));
+
+  bool callback_called = false;
+
+  ledger::PromotionPtr promotion = ledger::Promotion::New();
+  promotion->id = "ABC123";
+  promotion->suggestions = 1;
+
+  ledger::ResultCallback callback =
+      std::bind(
+        [&callback_called](ledger::Result result) {
+          callback_called = true;
+        }, _1);
+
+  const std::string payment_id = "bob";
+  ON_CALL(*mock_ledger_impl_, GetPaymentId())
+      .WillByDefault(testing::ReturnRef(payment_id));
+
+  ledger::WalletInfoProperties wallet_info;
+  wallet_info.key_info_seed.push_back(1);
+  ON_CALL(*mock_ledger_impl_, GetWalletInfo())
+      .WillByDefault(testing::ReturnRef(wallet_info));
+
+  // Act
+  promotion_->ClaimTokens(promotion.Clone(), callback);
+
+  // Assert
+  EXPECT_FALSE(callback_called);
+}
+
+TEST_F(PromotionTest, LegacyPromotionIsNotOverwritten) {
+  const std::string payment_id = "this_is_id";
+  ON_CALL(*mock_ledger_impl_, GetPaymentId())
```

Comment:
let's keep this in SetUp and just add variable with default value which can be overwritten

Reply:
`GetPaymentId()` is only needed for this test, why add for all tests?

Reply:
that is not true, you can see that @masparrow uses it in multiple places

Reply:
He does but with different fake payment id's?

Reply:
that's why I said variable

Reply:
It is used in multiple places but not all, and to add a global variable for me does not make sense

Reply:
Having a global Setup method isn't viable for multiple tests - we should only provide global Setup's when they are truly global, otherwise we should set up per test which is why I moved the code.